### PR TITLE
add DefaultBrowserBottomSheetDialog

### PR DESCRIPTION
**Pull request by LukasPaczos**
Task/Issue URL: https://app.asana.com/0/72649045549333/1208944504536341/f

### Description

Adds a promotion dialog that advertises setting DuckDuckGo as the default browser.

The introduced dialog is not yet used anywhere and click listeners/handlers will be added in following PRs.

### Steps to test this PR
As an example, apply below diff which opens the bottom sheet when the fire button is clicked.

```diff
diff --git a/app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt b/app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt
index d0e5ef7f8..28abbd21f 100644
--- a/app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt
+++ b/app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt
@@ -118,6 +118,7 @@ import com.duckduckgo.app.browser.customtabs.CustomTabPixelNames
 import com.duckduckgo.app.browser.customtabs.CustomTabViewModel.Companion.CUSTOM_TAB_NAME_PREFIX
 import com.duckduckgo.app.browser.databinding.FragmentBrowserTabBinding
 import com.duckduckgo.app.browser.databinding.HttpAuthenticationBinding
+import com.duckduckgo.app.browser.defaultbrowsing.prompts.ui.DefaultBrowserBottomSheetDialog
 import com.duckduckgo.app.browser.downloader.BlobConverterInjector
 import com.duckduckgo.app.browser.favicon.FaviconManager
 import com.duckduckgo.app.browser.filechooser.FileChooserIntentBuilder
@@ -2368,7 +2369,10 @@ class BrowserTabFragment :
                 }
 
                 override fun onFireButtonPressed(isPulseAnimationPlaying: Boolean) {
-                    onOmnibarFireButtonPressed(isPulseAnimationPlaying)
+                    // onOmnibarFireButtonPressed(isPulseAnimationPlaying)
+
+                    val dialogDefault = DefaultBrowserBottomSheetDialog(requireContext())
+                    dialogDefault.show()
                 }
 
                 override fun onBrowserMenuPressed() {
```

- [ ] Test portrait
- [ ] Test landscape and ensure all content can be accessed when scrolled.

### UI changes
| Light  | Dark |
| ------ | ----- |
Screenshot_20241213_121320|Screenshot_20241213_121339|

Video of usage in landscape:

https://github.com/user-attachments/assets/0a024142-42ac-4cc7-a6e6-f5062dffbd8d

### CODE REVIEW

File: saved-sites/saved-sites-impl/src/main/res/values/strings-saved-sites.xml [hunk 1]
```diff
@@ -107,4 +107,11 @@
     <string name="deleteBookmarkFolderConfirmationDialogDescription" instruction="Placeholder is the name of the favorite site">This will delete your &lt;b>%1$s&lt;/b> bookmark folder.</string>
     <string name="deleteSavedSiteConfirmationDialogCancel">Cancel</string>
     <string name="deleteSavedSiteConfirmationDialogDelete">Delete</string>
+
+    <!--    Default Browser Bottom Sheet Dialog -->
```
Comment (malmstein) [inline saved-sites/saved-sites-impl/src/main/res/values/strings-saved-sites.xml:111]: Did you mean adding the strings to the saves-sites module? I think they belong in the app module, like the rest of the code.

Also, as a good practice we want to add all the strings initially to a donottranslate.xml file. We do this until all the feature is completed, so we avoid generating translations. I’ll reach out to give you more context.

File: saved-sites/saved-sites-impl/src/main/res/drawable/ic_device_mobile_default_128.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (malmstein) [inline saved-sites/saved-sites-impl/src/main/res/drawable/ic_device_mobile_default_128.xml:1]: does this belong to the saved-sites module?

**Review by malmstein — APPROVED** [review]
Couple nits about file location and translations. Rest looks good @LukasPaczos !

File: saved-sites/saved-sites-impl/src/main/res/values/strings-saved-sites.xml (hunk 1 above)
Comment (LukasPaczos) [reply saved-sites/saved-sites-impl/src/main/res/values/strings-saved-sites.xml:111]: No, thanks for pointing this out!

**Comment by LukasPaczos** [comment]
Ready for another round!

**Pull request merged**